var/spack/repos/builtin/packages/libmesh/package.py

```diff
@@ -19,29 +19,182 @@ class Libmesh(Package):
+    # TODO libMesh 1.2.1 gained the ability to specify a path to capnproto
+    variant('capnproto', default=False, description='Compile with the bundled capnproto serialization library')
+    variant('exodusii', default=False, description='Compile with the bundled ExodusII output library')
+    variant('fparser', default=False, description='Compile with the bundled fparser function parser library')
+    variant('gmv', default=False, description='Compile with the bundled gmv format IO library')
+    variant('laspack', default=False, description='Compile with the bundled laspack interative solver library')
+    variant('libhilbert', default=False, description='Compile with the bundled libHilbert partitioning library')
+    variant('metaphysicl', default=False, description='Compile with the bundled metaphysicl AD library')
+    variant('metis', default=False, description='Compile with the bundled METIS graph partitioning library')
+    variant('nanoflann', default=False, description='Compile with the bundled nanoflann graph library')
+    variant('nemesis', default=False, description='Compile with the bundled nemesis IO library')
+    variant('netcdf', default=False, description='Compile with the bundled NETCDF IO library')
+    # TODO parmetis assumes that we use the bundled copy of metis, but the
+    # existing build script assumes that we can get METIS from PETSc. Be
+    # conservative and completely disable parmetis support for now.
+    # variant('parmetis', default=False,
+    #         description='Compile with the bundled PARMETIS graph library')
+    variant('qhull', default=False, description='Compile with the bundled qhull mesh generation library')
+    variant('sfc', default=False, description='Compile with the bundled sfcurves partitioning library')
+    variant('tetgen', default=False, description='Compile with the bundled tetgen mesh generation library')
+    variant('triangle', default=False, description='Compile with the bundled Triangle mesh generation library')
+
+    # support for libraries that may be externally installed:
+    variant('boost', default=False, description='Compile with components dependent on boost')
+    variant('eigen', default=False, description='support for dense linear algebra with Eigen')
+    variant('hdf5', default=False, description='Compile with support for HDF5 files')
+    variant('slepc', default=False, description='Compile with support for the SLEPc eigensolver')
+
+    # other features:
+    variant('debug', default=False, description='Compile with support for debugging')
     variant('mpi', default=True, description='Enables MPI parallelism')
-    variant('slepc', default=False, description='SLEPc eigensolver')
+    variant('openmp', default=False,
+            description='Enable OpenMP support; '
+            'this is independent of the choice of threading library')
+    variant('threads', default='none', description='Enable threading support',
```

Comment:
Why are there both threads and openmp variants? I would expect openmp to be here as one of the threads option

Reply:
I thought so at first as well, but libMesh allows one to enable openMP support independently of the threading support provided by either pthreads or tbb. I have not used any of them, but the configure code (but not the comments, unfortunately) is pretty clear on their (lack of) interdependence.

Reply:
Actually, even though one can enable openMP *and* one of either pthreads or tbb, I don't think that the libMesh configure script supports enabling openMP and *neither* threading library, even though the code supports it. I'll take a closer look.

Reply:
Hm, so you can enable tbb and openmp simultaneously? That sounds strange... say MKL allows you to use tbb for shared memory parallelization instead of openmp, but not both simultaneously.

It’s up to you, but I would keep openmp here within this variant despite the possibility to enable both simultaneously. I don’t recall how we call it in mkl/openblas/atlas, but I would use the same names.

Reply:
Clarification: the only time libMesh uses OpenMP (outside of some initialization) is in a header that is only used if pthread support is enabled.

Reply:
I implemented the proposed variant (i.e., switch between openMP and pthreads, pthreads alone, and TBB).